### DISCUSSION TITLE
Workflow status badge per branch

### DISCUSSION THREAD

**Discussion by localheinz** [community/community#25563] [Actions]
<p>According to <a href="https://help.github.com/en/articles/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository" rel="nofollow noopener">https://help.github.com/en/articles/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository</a> it is possible to add a status badge for a workflow.</p>
<p>Currently the documentation does not mention whether it is possible to reference a branch name. This is important (and useful) for projects that have long-living branches in addition to the default branch.</p>
<p>For example, <a href="https://github.com/sebastianbergmann/phpunit" rel="nofollow noopener">sebastianbergmann/phpunit</a> has multiple long-living branches</p>
<p>- <a href="https://github.com/sebastianbergmann/phpunit/tree/master" rel="nofollow noopener">master</a></p>
<p>- <a href="https://github.com/sebastianbergmann/phpunit/tree/8.3" rel="nofollow noopener">8.3</a></p>
<p>- <a href="https://github.com/sebastianbergmann/phpunit/tree/7.5" rel="nofollow noopener">7.5</a></p>
<p>As of the moment, when using Travis CI (see <a href="https://travis-ci.org" rel="nofollow noopener">https://travis-ci.org</a>), we can specify the name of the branch branch build status badges:</p>
<pre><code>[![Build Status](https://travis-ci.org/sebastianbergmann/phpunit.svg?branch=master)](https://travis-ci.org/sebastianbergmann/phpunit)
[![Build Status](https://travis-ci.org/sebastianbergmann/phpunit.svg?branch=7.5)](https://travis-ci.org/sebastianbergmann/phpunit)
[![Build Status](https://travis-ci.org/sebastianbergmann/phpunit.svg?branch=8.3)](https://travis-ci.org/sebastianbergmann/phpunit)
</code></pre>
<p>It would be great if we could do the same with GitHub Action status badges, for example:</p>
<pre><code>[![CI Status](https://github.com/sebastianbergmann/phpunit/workflows/CI/badge.svg?branch=master)](https://github.com/sebastianbergmann/phpunit/actions)
[![CI Status](https://github.com/sebastianbergmann/phpunit/workflows/CI/badge.svg?branch=8.3)](https://github.com/sebastianbergmann/phpunit/actions)
[![CI Status](https://github.com/sebastianbergmann/phpunit/workflows/CI/badge.svg?branch=7.5)](https://github.com/sebastianbergmann/phpunit/actions)
</code></pre>
<p>Is that possible?</p>

**Comment by mscoutermarsh** [comment]
<p>Badges by branch currently aren’t available. They currently check the status on your default branch (usually master).</p>
<p>Definitely agree it’s important. It’s on our list to get this added. :sparkles:</p>

**Comment by djotaku** [comment]
<p>is this also why PR status badges don’t work? I’ve got a push badge and a PR badge and only the push one seems to be working.</p>

**Comment by mxie** [comment]
<p>Yes - this is correct.</p>

**Comment by VitoRomaniello** [comment]
<p>Are there any news about badges by branch?</p>

**Comment by oldskool** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/vitoromaniello">@vitoromaniello</a> wrote:</p>
</blockquote>
<blockquote>
<p>Are there any news about badges by branch?</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>According to the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#example-using-the-branch-parameter" rel="nofollow noopener">documentation</a> it should now be possible to add a “branch” query parameter to the URL to get a branch specific badge, so you’ll get something like:</p>
<pre><code>![](https://github.com/user/repo/workflows/Tests/badge.svg?branch=develop)
</code></pre>

**Reply by MMukundi** [reply]
Here's an updated link if anyone has this question now:
[Link to documentation](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge#using-the-branch-parameter)
